App/Nevermore.lua

```diff
@@ -15,49 +15,46 @@ assert(script.Name == "Nevermore", "Invalid script name. For Nevermore to work c
 assert(script.Parent == ReplicatedStorage, "Invalid parent. For Nevermore to work correctly, it should be " ..
 	"a ModuleScript named \"Nevermore\" parented to ReplicatedStorage")
 
---- Handles yielded operations by caching the retrieval process
-local function handleRetrieving(retrieving, func, argument)
-	assert(type(retrieving) == "table", "Error: retrieving must be a table")
-	assert(type(func) == "function", "Error: func must be a function")
-
-	local bindableEvent = Instance.new("BindableEvent")
-	local result
-	retrieving[argument] = function()
-		if result ~= nil and result then
-			return result
-		end
+local AsyncMemoizer = {}
+AsyncMemoizer.ClassName = "AsyncMemoizer"
+AsyncMemoizer.__index = AsyncMemoizer
 
-		bindableEvent.Event:Wait()
-		return result
-	end
+--- Caches single argument of an asyncronious function
```

Comment:
This could benefit from improved documentation.

```lua
--@param func A function which takes one argument and returns one result. This function will only be called once for each unique argument.
--@return A function which, when called, invokes the function & caches its result.
```

Reply:
Agreed!